packages/aws-client/src/S3.ts

```diff
@@ -673,22 +681,28 @@ export const listS3Objects = async (
  *
  * @static
  */
-export const listS3ObjectsV2 = async (params: AWS.S3.ListObjectsV2Request) => {
+export const listS3ObjectsV2 = async (
+  params: AWS.S3.ListObjectsV2Request
+): ListS3ObjectsV2Result => {
   // Fetch the first list of objects from S3
-  let listObjectsResponse = await s3().listObjectsV2(params).promise();
+  let listObjectsResponse = <ListObjectsV2Output>(
+    await s3().listObjectsV2(params).promise()
+  );
+
   let discoveredObjects = listObjectsResponse.Contents;
 
   // Keep listing more objects from S3 until we have all of them
   while (listObjectsResponse.IsTruncated) {
-    listObjectsResponse = await s3().listObjectsV2( // eslint-disable-line no-await-in-loop
+    // eslint-disable-next-line no-await-in-loop
+    listObjectsResponse = <ListObjectsV2Output>(await s3().listObjectsV2(
       // Update the params with a Continuation Token
       {
 
         ...params,
         ContinuationToken: listObjectsResponse.NextContinuationToken
       }
-    ).promise();
-    discoveredObjects = (discoveredObjects || []).concat(listObjectsResponse.Contents || []);
+    ).promise());
+    discoveredObjects = discoveredObjects.concat(listObjectsResponse.Contents);
```

Comment:
So we have confidence based on the SDK docs that the return here should never be undefined?

Reply:
I think so. It's the assumption that we've been working off of for the past few years.